src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,31 @@
+import Popup from "./Popup";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector, handleFormSubmit }) {
+    super(popupSelector);
+    this._handlFormSubmit = handleFormSubmit;
+    this._modalForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  renderLoading(isLoading, loadingText = "Saving...") {
+    if (isLoading) {
+      this._submitButton.textContent = loadingText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
+  }
+
+  submitBtnAction(handleForm) {
```

Comment:
Needs correction: The name of the function should begin with a verb, reflecting the action that the function performs. Next, the elements to which the action refers are described. `setSubnitAction` would be good.

Reply:
Thank you for that helpful information when it comes to naming! I took 2 months off and coming back, I definitely see the importance of naming functions or variables correctly.